src/data/chats/chat.rs

```diff
@@ -16,8 +16,14 @@ use super::model_loader::ModelLoader;
 
 pub type ChatID = u128;
 
-#[derive(Clone, Debug)]
-pub enum ChatTokenArrivalAction {
+#[derive(Debug)]
+pub struct ChatEntityAction {
+    pub chat_id: ChatID,
```

Comment:
This and the `DownloadFileAction` now have an `id` so the actions can be routed back to the entity that should handle the message. 

These ids are also used on widgets to detect if they should redraw or not (compared to the previous signal based approach that would redraw everything).